### DISCUSSION TITLE
psi隐私求交，隐私性疑问

### DISCUSSION THREAD

**Discussion by zh5715615** [secretflow/secretflow#656] [General]
官方隐私求交用例跑通了([https://www.secretflow.org.cn/docs/secretflow/latest/zh-Hans/tutorial/PSI_On_SPU](https://www.secretflow.org.cn/docs/secretflow/latest/zh-Hans/tutorial/PSI_On_SPU))，但是没能想明白是怎么实现入参隐私的，从代码中看到的（如下代码段），入参是明文的csv明文路径，打开这个文件，里面也是明文的。但是通过psi_csv隐私求交函数，就计算得到了交集结果，整个过程都是明文的，隐私性在哪啊，是不是psi_csv还能再拆开调用。里面有对两个csv文件的加密操作，求大佬们指教。

**Comment by zhanglei486** [comment]
<img width="859" alt="image" src="https://github.com/secretflow/secretflow/assets/47308800/47209901-05ff-485a-803c-26236c992369">
上面链接中的教程为了演示方便，将多个参与方使用一台机器进行模拟，实际部署使用环境参与方有自己的机器（节点），参与方的入参csv文件都在自己的机器上，每一方执行psi_csv读入自己的csv文件后，执行psi协议，最终得到交集结果。
可以参考secretflow的部署方式和psi的benchmark文档，了解执行过程：
https://www.secretflow.org.cn/docs/secretflow/latest/zh-Hans/getting_started/deployment
https://www.secretflow.org.cn/docs/secretflow/latest/zh-Hans/developer/benchmark/psi_benchmark
